tidewave-core/tests/server_tests.rs

```diff
@@ -0,0 +1,335 @@
+use axum::http::StatusCode;
+use tokio::net::TcpListener;
+use tokio::sync::oneshot;
+use tidewave_core::config::Config;
+use tidewave_core::server::serve_http_server_with_shutdown;
+
+// ============================================================================
+// Test Helpers
+// ============================================================================
+
+/// Start a test HTTP server with configurable allowed origins
+/// Returns (port, shutdown_sender)
+async fn start_test_server(allowed_origins: Vec<String>) -> (u16, oneshot::Sender<()>) {
+    let config = Config {
+        port: 0, // Random port
+        allowed_origins,
+        ..Default::default()
+    };
+
+    // Bind the server to get the port
+    let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let port = listener.local_addr().unwrap().port();
+
+    // Create a shutdown channel
+    let (shutdown_tx, shutdown_rx) = oneshot::channel();
+
+    // Start the server in a background task
+    tokio::spawn(async move {
+        serve_http_server_with_shutdown(config, listener, async {
+            shutdown_rx.await.ok();
+        })
+        .await
+        .unwrap();
+    });
+
+    // Give the server a moment to start
+    tokio::time::sleep(tokio::time::Duration::from_millis(200)).await;
```

Comment:
hrm I mean it's not a lot, but it still feels bad to slow down tests here. Maybe we keep the which tests using the unit test approach?